src/FromFile.jl

```diff
@@ -12,23 +13,62 @@ function __init__()
     @require Revise="295af30f-e4ad-537b-8983-00126c2a3abe" track(mod, path) = Revise.track(mod, path)
 end
 
-macro from(path::String, ex::Expr)
+macro from(path, ex::Expr)
     esc(from_m(__module__, __source__, path, ex))
 end
 
-macro from(path::String)
-    esc(from_m(__module__, __source__, path, Expr(:block))) 
+macro from(path)
+    esc(from_m(__module__, __source__, path))
 end
 
-function from_m(m::Module, s::LineNumberNode, path::String, root_ex::Expr)
-    import_exs = if root_ex.head === :block
-        filter(ex -> !(ex isa LineNumberNode), root_ex.args)
-    else
-        [root_ex]
+macro from_url(path, ex::Expr)
+    esc(from_url_m(__module__, __source__, path, ex))
+end
+
+macro from_url(path)
+    esc(from_url_m(__module__, __source__, path))
+end
+
+# NOTE: we'd like to keep this pacakge minimal deps
+# so just copy paste this one-liner
+# Source: IsUrl.jl (MIT license)
+function isurl(str::AbstractString)
+    windowsregex = r"^[a-zA-Z]:[\\]"
+    urlregex = r"^[a-zA-Z][a-zA-Z\d+\-.]*:"
+    return !occursin(windowsregex, str) && occursin(urlregex, str)
+end
+
+function from_m(m::Module, s::LineNumberNode, path, root_ex::Expr = Expr(:block))
+    path = eval_quoted_string_expr(m, path)
+    return from_local_file(m, s, path, root_ex)
+end
+
+function from_url_m(m::Module, s::LineNumberNode, url, root_ex::Expr = Expr(:block))
+    url = eval_quoted_string_expr(m, url)
+    isurl(url) || throw(ArgumentError("expects an URL"))
+    return from_remote_file(m, s, url, root_ex)
+end
+
+function eval_quoted_string_expr(m::Module, str)
+    str isa String && return str
+    Meta.isexpr(str, :string) || throw(ArgumentError("expect a String, got $str"))
+    tokens = map(str.args) do s
+        s isa String && return s
+        return Base.eval(m, s)
     end
+    return join(tokens)
+end
```

Comment:
If I understand this correctly, it means we cannot do:

```julia
myfile = "file.jl"
@from myfile import foo
```
and must instead do:
```julia
myfile = "file.jl"
@from "$myfile" import foo
```
which is a bit weird.

Reply:
I think this is more consistent with our original syntax, in the original syntax string literal is allowed but interpolation is not, which is strange. And this syntax guarantees it's a string, not something else, no matter what is interpolated it will be a `String`